pkg/controller/deployment/deployment_controller.go

```diff
@@ -419,17 +427,45 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
 }
 
-func (dc *DeploymentController) syncRecreateDeployment(deployment extensions.Deployment) error {
-	newRC, err := dc.getNewRC(deployment)
+// Rolling back to a revision; no-op if the toRevision is deployment's current revision
+func (dc *DeploymentController) rollback(deployment extensions.Deployment, toRevision int) error {
+	if toRevision < 0 {
+		return nil
+	}
+	newRC, allOldRCs, err := dc.getNewRCAndAllOldRCs(deployment)
 	if err != nil {
 		return err
 	}
+	allRCs := append(allOldRCs, newRC)
+	// If rollback revision is 0, rollback to the last revision
+	if toRevision == 0 {
+		toRevision = secondMaxRevision(allRCs)
+	}
+	for _, rc := range allRCs {
+		v, err := revision(rc)
+		if err != nil {
+			return fmt.Errorf("unable to extract revision from deployment's RCs: %v", err)
```

Comment:
I would log this error and error out only in case the desired revision wasn't found